<launch>
    <include file = "$(find localization)/launch/racecar_model.launch"/>
    <include file = "$(find zed_wrapper)/launch/zed.launch"/>

    <node pkg="final_challenge" type="pure_pursuit.py" name="Trajectory_follower">
		<param name="odom_topic" value="/pf/pose/odom"/>
		<param name="drive_topic" value="/vesc/high_level/ackermann_cmd_mux/input/nav_3"/>
		<param name="speed" value="0.5"/>
		<param name="lookahead" value="0.5"/>
	</node>
    <node pkg="final_challenge" type="pure_turn.py" name="Pure_turner">
		<param name="turn_topic" value="/vesc/high_level/ackermann_cmd_mux/input/nav_2"/>
		<param name="speed" value="0.5"/>
	</node>

	<node pkg="final_challenge" type="city_driving_state_machine.py" name="state_machine">
		<param name="stop_detected" value="/stop_detected"/>
		<param name="car_wash_detected" value="/car_wash_detected"/>

		<!-- TOPICS THAT INDICATE STATE -->
		<param name="line_follower" value="/line_follower"/>
		<param name="stop_state" value= "/stop_state"/>
		<param name="car_wash_state" value="/car_wash_state"/>
	</node>

	<node pkg="final_challenge" type="orange_line_detector.py" name="Trajectory_planner" output="screen">
		<param name="image_topic" value="/zed/zed_node/rgb/image_rect_color"/>
		<param name="active_state" value="/line_follower"/>

		<!-- DEFAULTS TO BOOLEAN FALSE IN SCRIPT, I JUST HAVE THIS HERE AS A REMINDER THAT IT EXISTS -->
		<param name="start_active" value="True"/> 
	</node>

	<node pkg="final_challenge" type="stop_sign_controls.py" name="stop_detector_controller" output="screen">
	</node>


	<node pkg="wall_follower" type="safety_controller.py" name="safety_controller">
	</node>

</launch>
